### DISCUSSION TITLE
System.InvalidOperationException: Cannot perform bulk operation. Index is stale.

### DISCUSSION THREAD

**Discussion by TheMadKow** [ravendb/ravendb#13759] [Q&A] [answered]
Hey,

I'm getting the following error (see below) 
I'm using streams to get data and write the results back as bulk.

Where should I look for stale indexes? 

Best,
Eliran.

Error: 

System.InvalidOperationException: Cannot perform bulk operation. Index is stale. at Raven.Server.Documents.Queries.AbstractQueryRunner.ExecuteOperation[T](IndexQueryServerSide query, Index index, QueryOperationOptions options, QueryOperationContext queryContext, Action`1 onProgress, Func`3 createCommandForId, OperationCancelToken token) in C:\Builds\RavenDB-5.2-Cloud\52023\src\Raven.Server\Documents\Queries\AbstractQueryRunner.cs:line 174 at Raven.Server.Documents.Queries.Dynamic.DynamicQueryRunner.ExecutePatchQuery(IndexQueryServerSide query, QueryOperationOptions options, PatchRequest patch, BlittableJsonReaderObject patchArgs, QueryOperationContext queryContext, Action`1 onProgress, OperationCancelToken token) in C:\Builds\RavenDB-5.2-Cloud\52023\src\Raven.Server\Documents\Queries\Dynamic\DynamicQueryRunner.cs:line 114 at Raven.Server.Documents.Queries.QueryRunner.ExecutePatchQuery(IndexQueryServerSide query, QueryOperationOptions options, PatchRequest patch, BlittableJsonReaderObject patchArgs, QueryOperationContext queryContext, Action`1 onProgress, OperationCancelToken token) in C:\Builds\RavenDB-5.2-Cloud\52023\src\Raven.Server\Documents\Queries\QueryRunner.cs:line 398

**Comment by ppekrol** [comment, marked as answer]
Hi,

There are 3 types of queries in RavenDB:
1. Collection queries: `from Companies`
2. Dynamic queries: `from Companies where Name = "HR"`
3. Static Index queries: `from index 'Companies/ByName` where Name = "HR"`

If you are executing Patch by Query operation and it is 2nd or 3rd option, then either Auto or Static index will be matched in order to perform the patching operation. Those indexes can be found in the Indexes View in the Studio.

There is also a parameter QueryOperationOptions that allows you to perform that patching operation on stale data.

Hope that this helps.

**Comment by TheMadKow** [comment]
Yeah, Nailed this one.
I've got patch request before that.. I'll try to wait for non stale results =)

**Comment by DejanMilicic** [comment]
Here is an overview of staleness, how to detect and circumvent it https://github.com/DejanMilicic/RavenWorkshop/blob/main/Northwind/Northwind/Features/Staleness/Staleness.cs
